### ISSUE TITLE
Early stopping callback will stop early even if improvement when patience = 0

### ISSUE THREAD

**Issue by cnd9** [keras-team/keras#17305]
keras==2.7.0

The early stopping callback source code https://github.com/keras-team/keras/blob/v2.11.0/keras/callbacks.py#L1942-L2114 has the functionality where self.wait is set to zero if improvement occurs (L2080), but then in L 2083 has a check for stopping if `self.wait >= self.patience`, meaning (due to the greater than OR equal in L2093) the training will always stop early if patience = 0, even if the monitored metric is improving. 

This conflicts with the description of patience as the minimum number of epochs to wait if improvement does *not* occur.  It seems that L2083 should have a > operator and not >=.

**Comment by sushreebarsa** [comment]
@cnd9 Could you please check this [PR](https://github.com/keras-team/keras/pull/17366), and let us know if it helps?
Thank you!
